File: src/main/java/net/sf/rails/game/model/RailsModel.java
package net.sf.rails.game.model;

import net.sf.rails.game.RailsItem;
import net.sf.rails.game.RailsRoot;
import net.sf.rails.game.state.Model;

/**
 * RailsModel adds Rails specific methods to Model
 */
public class RailsModel extends Model implements RailsItem {

    protected RailsModel(RailsItem parent, String id) {
        super(parent, id);
    }
    
    @Override
    public RailsItem getParent() {
        return (RailsItem)super.getParent();
    }
    
    @Override
    public RailsRoot getRoot() {
        return (RailsRoot)super.getRoot();
    }
}

File: dotnet/RailsLib.Net/Net/Game/Model/RailsModel.cs
using GameLib.Net.Game.State;
using System;

/**
 * RailsModel adds Rails specific methods to Model
 */

namespace GameLib.Net.Game.Model
{
    public class RailsModel : State.Model, IRailsItem
    {
        protected RailsModel(IRailsItem parent, String id) : base(parent, id)
        {
        }

        new public IRailsItem Parent
        {
            get
            {
                return (IRailsItem)base.Parent;
            }
        }

        new public RailsRoot GetRoot
        {
            get
            {
                return (RailsRoot)base.GetRoot;
            }
        }
    }
}
